Source/Scene/ImplicitSubtree.js

```diff
@@ -0,0 +1,470 @@
+ * @private
+ */
+function requestActiveBuffers(subtree, bufferHeaders, internalBuffer) {
+  var promises = [];
+  for (var i = 0; i < bufferHeaders.length; i++) {
+    var bufferHeader = bufferHeaders[i];
+    if (!bufferHeader.isActive) {
+      promises.push(when.resolve(undefined));
+    } else if (bufferHeader.isExternal) {
+      var resource = subtree._resource.getDerivedResource({
+        url: bufferHeader.uri,
+      });
+      var promise = resource.fetchArrayBuffer().then(function (arrayBuffer) {
+        return new Uint8Array(arrayBuffer);
+      });
+      promises.push(promise);
+    } else {
+      promises.push(when.resolve(internalBuffer));
+    }
+  }
+  return when.all(promises).then(function (bufferResults) {
+    var buffersU8 = {};
+    for (var i = 0; i < bufferResults.length; i++) {
+      var result = bufferResults[i];
+      if (defined(result)) {
+        buffersU8[i] = result;
+      }
+    }
+    return buffersU8;
+  });
+}
+
+/**
+ * Go through the list of buffer views, and if they are marked as active,
+ * extract a subarray from one of the active buffers.
+ *
+ * @param {BufferViewHeader[]} bufferViewHeaders
+ * @param {Object} buffersU8 A dictionary of buffer index to a Uint8Array of its contents.
+ * @returns {Object} A dictionary of buffer view index to a Uint8Array of its contents.
+ * @private
+ */
+function parseActiveBufferViews(bufferViewHeaders, buffersU8) {
+  var bufferViewsU8 = {};
+  for (var i = 0; i < bufferViewHeaders.length; i++) {
+    var bufferViewHeader = bufferViewHeaders[i];
+
+    if (!bufferViewHeader.isActive) {
+      continue;
+    }
+
+    var start = bufferViewHeader.byteOffset;
+    var end = start + bufferViewHeader.byteLength;
+    var buffer = buffersU8[bufferViewHeader.buffer];
+    var bufferView = buffer.subarray(start, end);
+    bufferViewsU8[i] = bufferView;
+  }
+  return bufferViewsU8;
+}
+
+/**
+ * Parse the three availability bitstreams and store them in the subtree
+ *
+ * @param {ImplicitSubtree} subtree The subtree to modify
+ * @param {Object} subtreeJson The subtree JSON
+ * @param {ImplicitTileset} implicitTileset The implicit tileset this subtree belongs to
+ * @param {Object} bufferViewsU8 A dictionary of buffer view index to a Uint8Array of its contents.
+ * @private
+ */
+function parseAvailability(
+  subtree,
+  subtreeJson,
+  implicitTileset,
+  bufferViewsU8
+) {
+  var branchingFactor = implicitTileset.branchingFactor;
+  var subtreeLevels = implicitTileset.subtreeLevels;
+  var tileAvailabilityBits =
+    (Math.pow(branchingFactor, subtreeLevels) - 1) / (branchingFactor - 1);
+  var childSubtreeBits = Math.pow(branchingFactor, subtreeLevels);
+
+  subtree._tileAvailability = parseAvailabilityBitstream(
+    subtreeJson.tileAvailability,
+    bufferViewsU8,
+    tileAvailabilityBits
+  );
+
+  subtree._contentAvailability = parseAvailabilityBitstream(
+    subtreeJson.contentAvailability,
```

Comment:
`contentAvailability` could be undefined in the future once we add support for `3DTILES_multiple_contents`. Is it worth handling that situation now? Maybe it creates a dummy `ImplicitAvailabilityBitstream` with `constant: 0`

Reply:
I already handle defaulting to 0 in `initialize`

Reply:
I see now 👍